### PR TITLE
docs: Add Open Collective to README.rst

### PR DESCRIPTION
**Pull request by monkeywithacupcake**
Hi, I'm making updates for Open Collective. Either you or a supporter signed this repo up for Open Collective. This pull request adds backers and sponsors from your Open Collective https://opencollective.com/sopel❤️

It adds two badges at the top to show the latest number of backers and sponsors. It also adds placeholders so that the avatar/logo of new backers/sponsors can automatically be shown without having to update your README. [more info](https://github.com/opencollective/opencollective/wiki/Github-banner). See how it looks on this [repo](https://github.com/apex/apex#backers).

You can also add a postinstall script to let people know after npm|yarn install that you are welcoming donations (optional). [More info](https://github.com/OpenCollective/opencollective-cli)
You can also add a "Donate" button to your website and automatically show your backers and sponsors there with our widgets. Have a look here: https://opencollective.com/widgets

P.S: As with any pull request, feel free to comment or suggest changes. The only thing "required" are the placeholders on the README because we believe it's important to acknowledge the people in your community that are contributing (financially or with code!).

Thank you for your great contribution to the open source community. You are awesome! 🙌
And welcome to the open collective community! 😊

Come chat with us in the #opensource channel on https://slack.opencollective.com - great place to ask questions and share best practices with other open source sustainers!

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -57,6 +57,39 @@ Questions?
...
+Credits
++++++++
+
+Contributors
+------------
+
+This project exists thanks to all the people who contribute! `Become a contributor`__.
+
+.. image:: https://opencollective.com/sopel/contributors.svg?width=890&button=false
+    :target: https://github.com/evhub/sopel/graphs/contributors
```
Comment (dgw) [inline README.rst]: Where did this repo come from? It doesn't exist.

Comment (monkeywithacupcake) [reply README.rst]: possibly a typo by me. I try not to start over with each RST because I usually write MD, so I copy the last one.

Comment (dgw) [inline README.rst]:
```suggestion
    :target: https://github.com/sopel-irc/sopel/graphs/contributors
```

Comment (dgw) [reply README.rst]:
```suggestion
    :target: https://github.com/sopel-irc/sopel/graphs/contributors
```

**Comment by dgw** [comment]
Sopel's domain name is now mine to maintain, so this will become more relevant as time goes on.

As it is, though, the document does not render. Having a readme that renders properly on both GitHub and PyPI is important to me (and, I think, to the project making a good impression on visitors), so whatever the issue is, it needs fixing. @monkeywithacupcake are you up for it, or should I investigate the problem on my own when I have a chance? I'd like to know what this PR is supposed to look like. :smile_cat:

**Comment by monkeywithacupcake** [comment]
I think I fixed the link for you

**Comment by dgw** [comment]
I really ought to remember to use `[Skip CI]` when I make documentation-only commits…
Poor overworked Travis!

Used http://rst.ninjs.org/ to track down the rendering errors. Definitely going to still play with the layout some more, since there are now too many badges to fit alongside the `Introduction` header. Maybe we'll get a proper header with a logo and everything out of this, too. :smirk_cat:

**Comment by dgw** [comment]
Rebased to clean up the commit messages a bit.

When did GitHub add that timeline event for force-pushes? That's neat. Means one can't stealth-update a broken PR any more though! :joy_cat:

**Comment by dgw** [comment]
Decided not to put a logo in the readme for now, but it does look quite nice (on GitHub, at least) with the badges moved around. Hopefully it looks good on PyPI too.

**Pull request merged**